src/strings.cpp

```diff
@@ -1478,7 +1478,7 @@ static void FormatString(StringBuilder &builder, const char *str_arg, StringPara
 				} else if (_scan_for_gender_data) {
 					/* Gender is defined by the industry type.
 					 * STR_FORMAT_INDUSTRY_NAME may have the town first, so it would result in the gender of the town name */
-					auto tmp_params = AllocatedStringParameters();
+					auto tmp_params = ArrayStringParameters<0>();
```

Comment:
Is a zero length array even valid? Bit strange regardless...

Reply:
The specification specifically mention a zero length array, and how it should behave. I'd say it's better to leave the special case in the standard library, than having to implement a special string parameters class for when there are no parameters at all.